modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/HeavyOrLightWeightAppControlTestCase.java

```diff
@@ -0,0 +1,204 @@
+        .withGroupId(MULE_CONNECTORS_GROUP_ID)
+        .withVersion("1.2.0")
+        .withClassifier(MULE_PLUGIN_CLASSIFIER);
+
+    final ApplicationFileBuilder applicationFileBuilder = appFileBuilder(LIGHTWEIGHT_APP)
+        .definedBy(APP_XML_FILE)
+        .dependingOn(mulePlugin)
+        .usingLightWeightPackage();
+
+    addPackedAppFromBuilder(applicationFileBuilder);
+    startDeployment();
+
+    final String applicationName = applicationFileBuilder.getDeployedPath();
+    final File applicationRepository = Paths.get(getAppFolder(applicationName).toString(), "repository").toFile();
+    final File muleRepository = Paths.get(getMuleBaseFolder().getAbsolutePath(), "repository").toFile();
+    final List<String> logCauseMessages = getLogCauseMessages(logger.getAllLoggingEvents());
+    final String exceptionError =
+        "org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.mule.connectors:mule-sockets-connector";
+
+    assertDeploymentFailure(applicationDeploymentListener, applicationFileBuilder.getId());
+    assertApplicationAnchorFileDoesNotExists(applicationName);
+    assertThat(logCauseMessages, hasItem(startsWith(exceptionError)));
+    assertThat(applicationRepository.exists(), is(false));
+    assertThat(muleRepository.exists(), is(false));
+  }
+
+  @Test
+  @Story(HEAVYWEIGHT)
+  @Description("Test case for MULE-12298 and MULE-12317")
+  public void heavyweightAppDeploymentDoesntDownloadDependenciesFromRemoteRepo() throws Exception {
+    final ArtifactPluginFileBuilder mulePlugin = new ArtifactPluginFileBuilder("mule-http-connector")
+        .withGroupId(MULE_CONNECTORS_GROUP_ID)
+        .withVersion("1.6.1")
+        .withClassifier(MULE_PLUGIN_CLASSIFIER)
+        .configuredWith(EXPORTED_CLASS_PACKAGES_PROPERTY, "org.foo");
+
+    final ApplicationFileBuilder applicationFileBuilder = appFileBuilder(HEAVYWEIGHT_APP)
+        .definedBy(APP_XML_FILE)
+        .dependingOn(mulePlugin);
+
+    addPackedAppFromBuilder(applicationFileBuilder);
+    startDeployment();
+
+    final String applicationName = applicationFileBuilder.getDeployedPath();
+    final File applicationRepository = Paths.get(getAppFolder(applicationName).toString(), "repository").toFile();
+    final File muleRepository = Paths.get(getMuleBaseFolder().getAbsolutePath(), "repository").toFile();
+    final List<String> applicationRepositoryContentNames =
+        getRepositoryContents(applicationRepository).stream().map(File::getName).collect(toList());
+
+    assertDeploymentSuccess(applicationDeploymentListener, applicationFileBuilder.getId());
+    assertApplicationAnchorFileExists(applicationName);
+    assertThat(applicationRepository.exists(), is(true));
+    assertThat(applicationRepositoryContentNames,
+               hasItem(allOf(startsWith("mule-http-connector"), endsWith("-mule-plugin.jar"))));
+    assertThat(muleRepository.exists(), is(false));
+  }
+
+  @NotNull
```

Comment:
This annotation is causing Hawk to fail, is it really necessary? Maybe you can find a Java annotation that does the same (instead of using `org.jetbrains.annotations`)

Reply:
Done.